source/Camera/Basic/C++Halcon/HalconConnectUsingGenICam/HalconConnectUsingGenICam.cpp

```diff
@@ -0,0 +1,117 @@
+#include <Zivid/Zivid.h>
+#include <halconcpp/HalconCpp.h>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include <chrono>
+#include <iostream>
+
+
+void savePointCloud(const HalconCpp::HObjectModel3D &model, const std::string filename) {
+	model.WriteObjectModel3d(HalconCpp::HString{ "ply" },
+		HalconCpp::HString{ filename.c_str() },
+		HalconCpp::HString{ "invert_normals" },
+		HalconCpp::HString{ "false" });
+}
+
+
+void setColorsInObjectModel3D(HalconCpp::HObjectModel3D &objectModel3D, const HalconCpp::HImage &RGB) {
+
+	HalconCpp::HRegion domain = RGB.AccessChannel(1).GetDomain();
+	HalconCpp::HTuple rows, cols;
+	domain.GetRegionPoints(&rows, &cols);
+
+	HalconCpp::HTuple Rtup = RGB.AccessChannel(1).GetGrayval(rows, cols);
+	HalconCpp::HTuple Gtup = RGB.AccessChannel(2).GetGrayval(rows, cols);
+	HalconCpp::HTuple Btup = RGB.AccessChannel(3).GetGrayval(rows, cols);
+
+	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("red"), "points", Rtup);
+	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("green"), "points", Gtup);
+	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("blue"), "points", Btup);
+
+}
+
+HalconCpp::HTuple getFirstAvailableZividDevice(const HalconCpp::HTuple &devices) {
+	auto device = HalconCpp::HTuple();
+	auto tmpSplit = HalconCpp::HTuple();
+	auto splitDevices = HalconCpp::HTuple();
+
+	for (int i = 0; devices.Length(); i++) {
+		HalconCpp::TupleSplit(devices[i], "|", &tmpSplit);
+		HalconCpp::TupleSplit(tmpSplit, ":", &splitDevices);
+
+		if (int(splitDevices.TupleSelect(2).TupleRegexpTest("Zivid"))) {
+			device = splitDevices.TupleSelect(2).TupleStrFirstN(splitDevices.TupleSelect(2).TupleStrlen() - 2);
+			break;
+		}
+	}
+	return device;
+}
+
+
+int main()
+{
+	try
+	{
+		std::cout << "Connecting to camera" << std::endl;
```

Comment:
Move this line above Framegrabber.OpenFramegrabber("Ge.... like in https://github.com/zivid/halcon-samples/blob/master/source/Camera/Basic/Capture.hdev

Reply:
This has not been resolved in the first iteration.

Reply:
Misunderstood you the first time, corrected now.